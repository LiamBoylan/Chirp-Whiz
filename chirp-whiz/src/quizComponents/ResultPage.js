import React from 'react';
import birdList from './../birdList';   // Bird info

/*
    This is the page that renders at the end of the quiz. It lets the user know how many
    questions they got correct and which questions in the quiz that they were.
    Props are the arrays of correct and incorrect questions for that quiz along with
    the total questions in the quiz.
*/

function ResultPage(props) {
    return (
        <div>
            <h1>You answered {Object.keys(props.correct).length} out of {props.totalQs} correct</h1>
            <br />
            <h3>Birds <span style={{color: 'green'}}>correctly</span> identified &#9989;:</h3>
            {Object.keys(props.correct).map(key => <p>{key}. {birdList[props.correct[key]].name}</p>)}
            <h3>Birds <span style={{color: 'red'}}>incorrectly</span> identified:</h3>
            {Object.keys(props.incorrect).map(key => <p>{key}. {birdList[props.incorrect[key]].name}</p>)}
            <br />
        </div>
    );
}

export default ResultPage;