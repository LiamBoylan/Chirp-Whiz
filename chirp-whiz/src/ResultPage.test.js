import React from 'react';
import ResultPage from './quizComponents/ResultPage';
import '@testing-library/jest-dom/extend-expect'
import { render, fireEvent, waitForElement } from '@testing-library/react'

// Integration tests

it('ResultPage snapshot', () => {
    const container = render(<ResultPage
        totalQs={5}
        correct={{}}
        incorrect={{}}
      />)
    expect(container.firstChild).toMatchSnapshot();
});

test("Test for if the results page accurately portrays empty results", () => {
    const {getByText} = render(<ResultPage
        totalQs={5}
        correct={{}}
        incorrect={{}}
      />)
    expect(getByText("You answered 0 out of 5 correct")).toBeInTheDocument();
});