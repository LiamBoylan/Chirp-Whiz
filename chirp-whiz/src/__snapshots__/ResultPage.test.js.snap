// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPage snapshot 1`] = `undefined`;
